### PR TITLE
[DEPRECATED] Philipye314/imp 11 eligibility

### PR DESCRIPTION
**Pull request by philipye314**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Created EligibilityCard component on caseScreen
- Created EligibilityForm component that allows users to update their eligibility

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
IMG_4657
IMG_4656
IMG_4655

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Add a new case and make sure you are getting the "CHECK ELIGIBILITY CARD" at first
- Update the eligibility to "YES I AM ELIGIBLE". Double check that you are now seeing the "FILE CLAIM" and "OPT OUT" cards.
- Do the same with a different case, update to "NO I AM NOT ELIGIBLE". Double check you will still get the "CHECK ELIGIBILITY" card.

## Next steps

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Need to fix typescript bug in cases.ts whoops
- Josh probably wants me to change the "FILE CLAIM" and "OPT OUT" buttons
- Temporary back button doesn't look very nice, doesn't make sense currently

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: .eslintignore 2
Comment (stephaniewong2) [inline .eslintignore 2:1]: Delete all extra eslintignore file! 😭

File: src/supabase/queries/cases.ts [hunk 1]
```diff
@@ -94,6 +122,48 @@ export function parseCase(item: any): Case {
...
+    const {
+      data: { user },
+    } = await supabase.auth.getUser();
+    const userId = user?.id;
+    const { data, error } = await supabase
+      .from('status')
+      .select()
+      .eq('userId', userId)
+      .eq('caseId', caseId);
+    return data[0].eligibility; //bruhhhh
```
Comment (stephaniewong2) [inline src/supabase/queries/cases.ts:162]: It errors because data can be null so add a check like `getCaseById`

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
Very very very brief review because I know this is a draft

File: src/Components/EligibilityCard/EligibilityCard.tsx [hunk 2]
```diff
@@ -1,10 +1,92 @@
...
+              params: { caseId: caseData.id },
+            });
+          }}
+        >
+          <View style={styles.topText}>
+            <Check />
+            <View style={styles.center}>
+              <Text style={styles.text}>Check Eligibility</Text>
+            </View>
+            <Arrow />
```
Comment (stephaniewong2) [inline src/Components/EligibilityCard/EligibilityCard.tsx:41]: Sorry this might be fault because of how i named svgs but can we use `next.svg` instead of the arrow!

File: src/Components/EligibilityCard/EligibilityCard.tsx [hunk 3]
```diff
@@ -1,10 +1,92 @@
...
+interface EligibilityCardProps {
+  caseData: Case;
+  status: Eligibility;
+}
+
+export default function EligibilityCard({
+  caseData,
+  status,
+}: EligibilityCardProps) {
+  const router = useRouter();
```
Comment (stephaniewong2) [inline src/Components/EligibilityCard/EligibilityCard.tsx]: Let's import `router` instead of `useRouter` for consistency and so that we don't need to declare `router` in every screen.

File: src/Components/EligibilityCard/EligibilityCard.tsx [hunk 4]
```diff
@@ -1,10 +1,92 @@
...
+            <View style={styles.topText}>
+              <Optout />
+              <View style={styles.center}>
+                <Text style={styles.text}>Opt Out</Text>
+              </View>
+              <Arrow />
+            </View>
+
+            <Text>
+              Submit a claim by July 14, 2023 if you wish to receive a payment
```
Comment (stephaniewong2) [inline src/Components/EligibilityCard/EligibilityCard.tsx]: Let's remove "by July 14, 2023" for now. We need to ask for clarifications here and if hardcoded, it's best to not hardcode a date

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import { View, Text, Image } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { useLocalSearchParams, useRouter } from 'expo-router';
+import styles from './styles';
+import { updateCaseStatus } from '../../../../../supabase/queries/cases';
+import { CaseUid, Eligibility } from '../../../../../types/types';
+import Arrow from '../../../../../../assets/arrow.svg';
+
+export default function EligibilityForm() {
+  const router = useRouter();
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx]: Same comment as above but let's import `router` instead of `useRouter` so we can remove this line

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const { caseId } = useLocalSearchParams<{
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx]: Let's use `const { caseId } = useLocalSearchParams() as unknown as { caseId: CaseUid };` this instead for consistency in how we use local params. We need to make sure we do error handling from the screen we are passing this down from to do this though.

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+          <View>
+            <Text>Do you meet the following requirements?</Text>
+          </View>
+        </View>
+      </View>
+      <View style={styles.buttonsContainer}>
+        <View style={styles.buttonWrapperTop}>
+          <TouchableOpacity
+            style={[styles.button, styles.buttonTop]}
+            onPress={() => updateCaseStatus(caseId, Eligibility.ELIGIBLE)}
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx:41]: `updateCaseStatus` is an async function so let's use a combination of `useEffect` and `useState` to update eligibility. We can have an `eligibility` state variable and a `useEffect` that calls `updateCaseStatus` and sets this variable!

Comment (stephaniewong2) [reply src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx:41]: I'm looking into routing with this right now and running into annoying issues (just wanted to let you know but don't worry about that right now). The only way I can confirm that this is correct right now is by going back to the `Cases` screen with the Case Cards and seeing that the "File Claim" and "Opt Out" buttons appear if the status is correct. This with a mix of logging.

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+          >
+            <Text>Yes, I am Eligible</Text>
+            <Arrow />
+          </TouchableOpacity>
+        </View>
+        <View style={styles.buttonWrapperBottom}>
+          <TouchableOpacity
+            style={[styles.button, styles.buttonBottom]}
+            onPress={() => updateCaseStatus(caseId, Eligibility.INELIGIBLE)}
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/EligibilityForm/index.tsx:50]: Same comment as above for `ELIGIBLE`

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx [hunk 9]
```diff
@@ -8,12 +8,21 @@ import CaseSummaryCard from '../../../../Components/CaseSummaryCard/CaseSummarCa
 import EducationalBar from '../../../../Components/EducationalBar/EducationalBar';
 import EligibilityCard from '../../../../Components/EligibilityCard/EligibilityCard';
 import FormsCard from '../../../../Components/FormsCard/FormsCard';
-import { Case } from '../../../../types/types';
+import { Case, Eligibility } from '../../../../types/types';
+import { getCaseStatus } from '../../../../supabase/queries/cases';
 
 function CasesScreen() {
   const caseData = useLocalSearchParams() as unknown as Case;
+  const [status, setStatus] = useState(Eligibility.ELIGIBLE);
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx:16]: Maybe we should set default to `UNDEFINED`

File: src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx [hunk 10]
```diff
@@ -8,12 +8,21 @@ import CaseSummaryCard from '../../../../Components/CaseSummaryCard/CaseSummarCa
...
 
-  console.log(caseData);
+  useEffect(() => {
+    const getStatus = async () => {
+      const caseStatus = await getCaseStatus(caseData.id);
+      setStatus(caseStatus);
+      console.log(caseStatus);
```
Comment (stephaniewong2) [inline src/app/(BottomTabNavigation)/Cases/CaseScreen/index.tsx]: Remove log statement

File: src/supabase/queries/cases.ts [hunk 11]
```diff
@@ -94,6 +122,48 @@ export function parseCase(item: any): Case {
...
+    const {
+      data: { user },
+    } = await supabase.auth.getUser();
+    const userId = user?.id;
+    const { error } = await supabase
+      .from('status')
+      .update({ eligibility: status })
+      .eq('userId', userId)
+      .eq('caseId', caseId);
+    console.log(error);
```
Comment (stephaniewong2) [inline src/supabase/queries/cases.ts]: error is caught in the try catch so we can remove this console.log here and log it in the `catch` statement

File: src/supabase/queries/cases.ts [hunk 12]
```diff
@@ -94,6 +122,48 @@ export function parseCase(item: any): Case {
...
+  }
+}
+
+export async function getCaseStatus(caseId: CaseUid) {
+  try {
+    const {
+      data: { user },
+    } = await supabase.auth.getUser();
+    const userId = user?.id;
+    const { data, error } = await supabase
```
Comment (stephaniewong2) [inline src/supabase/queries/cases.ts]: Let's remove `error` since we don't use it here

File: src/supabase/queries/cases.ts [hunk 13]
```diff
@@ -94,6 +122,48 @@ export function parseCase(item: any): Case {
...
+      .from('status')
+      .select()
+      .eq('userId', userId)
+      .eq('caseId', caseId);
+    return data[0].eligibility; //bruhhhh
+  } catch (error) {
+    throw error;
```
Comment (stephaniewong2) [inline src/supabase/queries/cases.ts:165]: Let's log the error like the other queries for consistency!

File: src/supabase/queries/cases.ts [hunk 14]
```diff
@@ -71,6 +71,34 @@ export async function getCasesByIds(caseIds: CaseUid[]): Promise<Case[]> {
   }
 }
 
+export async function uploadCase(caseId: CaseUid | undefined) {
```
Comment (stephaniewong2) [inline src/supabase/queries/cases.ts:74]: Sorry might've missed this but did you write these queries (`uploadCase` and `containsDuplicateCase`)?

**Review by stephaniewong2 — CHANGES_REQUESTED** [review]
Generally looks good! Very nice styling WOOOHOOO!! There are a few functionality issues that I have yet to figure out how to resolve but I will try to get back to you on Sunday (mostly just how to get screens to re-render properly with changes)

File: src/supabase/queries/cases.ts (hunk 14 above)
Comment (philipye314) [reply src/supabase/queries/cases.ts:74]: I believe so yes, but they would've been in a previous pr I'm pretty certain
